extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -428,7 +433,52 @@ private String getEntityName() {
     }
 
     private boolean entityContainsField(String fieldName) {
-        return entityClass.field(fieldName) != null;
+        if (entityClass.field(fieldName) != null) {
+            return true;
+        }
+
+        for (ClassInfo superClass : mappedSuperClassInfos) {
+            FieldInfo fieldInfo = superClass.field(fieldName);
+            if (fieldInfo != null) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private FieldInfo getField(String fieldName) {
+        FieldInfo fieldInfo = entityClass.field(fieldName);
+        if (fieldInfo == null) {
+            for (ClassInfo superClass : mappedSuperClassInfos) {
+                fieldInfo = superClass.field(fieldName);
+                if (fieldInfo != null) {
+                    break;
+                }
+            }
+        }
+        return fieldInfo;
+    }
+
+    private List<ClassInfo> getMappedSuperClassInfos(IndexView indexView, ClassInfo entityClass) {
+        List<ClassInfo> mappedSuperClassInfos = new ArrayList<>(3);
+        Type superClassType = entityClass.superClassType();
+        while (superClassType != null && !superClassType.name().toString().equals("java.lang.Object")) {
```

Comment:
`superClassType != null` isn't needed and `"java.lang.Object"` can be replaced by `DotNames.OBJECT`

Reply:
Of course `superClassType != null` is needed so the first part of the comment was dumb